python/MooseDocs/commands/MooseDocsMarkdownNode.py

```diff
@@ -63,8 +61,11 @@ def build(self):
 
     # Make sure the destination directory exists
     destination = self.path()
+    lock = multiprocessing.Lock()
+    lock.acquire()
     if not os.path.exists(destination):
       os.makedirs(destination)
+    lock.release()
 
```

Comment:
Maybe use the lock as a context manager so that if `makedirs` raises an exception then the lock will be released.